### PR TITLE
PricingSubcategory

### PR DESCRIPTION
**Pull request by flanakin**
Completing the breakdown of PricingCategory that we discussed over the past couple weeks.

Related #330

### CODE REVIEW

File: specification/columns/pricingsubcategory.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Pricing Subcategory
+
+Pricing Subcategory describes the pricing model used for a charge within a specific [Pricing Category](#pricingcategory). It can be useful for distinguishing between charges that share the same Pricing Category, but have different pricing rules that may result in a lower prices, like flat-rate compared to volume-based discounts.
```
Comment (SanjnaSrivatsaVMware) [inline specification/columns/pricingsubcategory.md:3]: Not entirely sure if the words "pricing model" are self-explanatory. Is there possibly another way to say that might be less 'left to interpretation' ?

Comment (flanakin) [reply specification/columns/pricingsubcategory.md:3]: Hmm... 🤔 Will put some thought into it. I used the same thing we have for PricingCategory, but we can update both places. Anyone else have thoughts or ideas?

File: specification/columns/commitmentdiscountcategory.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ Allowed values:
 |:--------|:-------------------------------------------------------------------------|
 | Spend   | Commitment-based discounts that require a predetermined amount of spend. |
 | Usage   | Commitment-based discounts that require a predetermined amount of usage. |
+| Other   | Commitment-based discounts that do not fall into another category.       |
```
Comment (cnharris10) [inline specification/columns/commitmentdiscountcategory.md:35]: Is this just to make sure the column remains applicable for the future?  I don't know of any commitment models that fall outside of "Usage" or "Spend" offhand.

File: specification/columns/pricingsubcategory.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Allowed values when PricingCategory is "On-Demand":
+
+| Value    | Description                                                                                                           |
+| :------- | :-------------------------------------------------------------------------------------------------------------------- |
+| Standard | On-demand charges priced at a flat rate for a SKU that does not support volume-based or tiered pricing.               |
+| Tiered   | On-demand charges priced at a specific volume-based or tiered level.                                                  |
+| Bundled  | On-demand charges priced at a free or reduced rate due to a companion SKU or service that was also used or purchased. |
+| Preview  | On-demand charges priced at a free or reduced rate for a prerelease version of the SKU being used or purchased.       |
+| Other    | On-demand charges priced in a way not covered above.                                                                  |
```
Comment (cnharris10) [inline specification/columns/pricingsubcategory.md:51]: Private Rates?

**Review by cnharris10 — CHANGES_REQUESTED** [review]

File: specification/columns/commitmentdiscountcategory.md (hunk 2 above)
Comment (flanakin) [reply specification/columns/commitmentdiscountcategory.md:35]: Correct, this is just about covering a new scenario that doesn't fit into the existing buckets. There are no other known commitment discount types today.

File: specification/columns/pricingsubcategory.md (hunk 3 above)
Comment (flanakin) [reply specification/columns/pricingsubcategory.md:51]: Is that the same as negotiated discounts? If so, that would still be "standard" pricing, just with reduced rates, right?

**Comment by udam-f2** [comment]
Let's discuss if there's a need for a pricing subcategory in v1.0 and re-open accordingly. This hasn't yet gone through the standard process of git issue, and discussion in TFs. If it was a minor change, that would be fine but for introducing a new column, we need to go through the standard process.

**Comment by flanakin** [comment]
This is part of #330. Let's not close it until we close that conversation please.

File: specification/columns/pricingsubcategory.md (hunk 1 above)
Comment (marc-perreaut) [reply specification/columns/pricingsubcategory.md:3]: I find that pricing model is a common concept, but to me the issue is PricingCategory and PricingSubcategory do not actually describe the pricing model but refer to categories and sub categories of used pricing models.

So maybe simply:
> Pricing Subcategory describes the sub category in the pricing model used for a charge within a specific Pricing Category

If the term pricing model is too abstract, maybe the term pricing type could be used instead.

File: specification/columns/pricingsubcategory.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+| Allows nulls    | True           |
+| Data type       | String         |
+| Value format    | Allowed values |
+
+Allowed values when PricingCategory is "On-Demand":
+
+| Value    | Description                                                                                                           |
+| :------- | :-------------------------------------------------------------------------------------------------------------------- |
+| Standard | On-demand charges priced at a flat rate for a SKU that does not support volume-based or tiered pricing.               |
+| Tiered   | On-demand charges priced at a specific volume-based or tiered level.                                                  |
```
Comment (marc-perreaut) [inline specification/columns/pricingsubcategory.md:48]: I see Tiered, Bundled, Preview as pricing dimensions, but not as pricing sub categories: they could combine together and apply to other pricing categories than On-Demand, even though I do not have an example in mind.

**Comment by flanakin** [comment]
Deferring to post-1.0

**Pull request closed without merging**